pkg/products/threescale/reconciler.go

```diff
@@ -815,48 +814,6 @@ func (r *Reconciler) reconcileUpdatingDefaultAdminDetails(ctx context.Context, s
 		return integreatlyv1alpha1.PhaseFailed, err
 	}
 
-	kcUsers := filterUsers(kcr.Spec.Users, func(u *aerogearv1.KeycloakUser) bool {
```

Comment:
Looks like the whole function (reconcileUpdatingDefaultAdminDetails) can be removed since it doesn't modify anything anymore.

Reply:
Good catch, it won't do anything now. 
But there is some editing of "system-seed" secret that I don't fully understand, maybe we need that logic..?
Let's create a follow up task, as this can be considered out of scope for this PR. wdyt?